File: snapshots/input/syntax/VBMain/Classes.vb
Imports System.Diagnostics.CodeAnalysis

Namespace VBMain
    <SuppressMessage("ReSharper", "all")>
    Public Class Classes

        Public Name As String
        Public Const IntConstant As Integer = 1
        Public Const StringConstant As String = "hello"

        Public Sub New(ByVal name As Integer)
            Me.Name = "name"
        End Sub

        Public Sub New(ByVal name As String)
            Me.Name = name
        End Sub

        Protected Overrides Sub Finalize()
            Console.WriteLine(42)
        End Sub

        Public Class ObjectClass
            Inherits Object
            Implements SomeInterface
        End Class

        Public Partial Class PartialClass
        End Class

        Class TypeParameterClass(Of T)
        End Class

        Friend Class InternalMultipleTypeParametersClass(Of T1, T2)
        End Class
        
        Interface ICovariantContravariant(Of In T1, Out T2)
        Sub Method1(ByVal t1 As T1)

        Function Method2() As T2

        End Interface

        Public Class StructConstraintClass(Of T As Structure)
        End Class

        Public Class ClassConstraintClass(Of T As Class)
        End Class

        Public Class NewConstraintClass(Of T As New)
        End Class

        Public Class TypeParameterConstraintClass(Of T As SomeInterface)
        End Class

        Private Class MultipleTypeParameterConstraintsClass(Of T1 As {SomeInterface, SomeInterface2, New}, T2 As SomeInterface2)
        End Class

        Class IndexClass
            Private a As Boolean

            Default Public Property Item(ByVal index As Integer) As Boolean
                Get
                    Return a
                End Get
                Set(ByVal value As Boolean)
                    a = value
                End Set
            End Property
        End Class

    Interface SomeInterface
    End Interface

    Friend Interface SomeInterface2
    End Interface

    End Class

End Namespace

File: snapshots/input/syntax/Main/Classes.cs
using System.Diagnostics.CodeAnalysis;

namespace Main;

[SuppressMessage("ReSharper", "all")]
public class Classes
{
    public string Name;
    public const int IntConstant = 1;
    public const string StringConstant = $"hello";

    public Classes(int name)
    {
        Name = "name";
    }

    public Classes(string name) => Name = name;

    ~Classes()
    {
        Console.WriteLine(42);
    }

    public class ObjectClass : object, SomeInterface
    {
    }

    public partial class PartialClass
    {
    }

    class TypeParameterClass<T>
    {
    }

    internal class InternalMultipleTypeParametersClass<T1, T2>
    {
    }

    interface ICovariantContravariant<in T1, out T2>
    {
        public void Method1(T1 t1)
        {
            Console.WriteLine(t1);
        }

        public T2? Method2()
        {
            return default(T2);
        }
    }

    public class StructConstraintClass<T> where T : struct
    {
    }

    public class UnmanagedConstraintClass<T> where T : unmanaged
    {
    }

    public class ClassConstraintClass<T> where T : class
    {
    }

    public class NonNullableConstraintClass<T> where T : notnull
    {
    }

    public class NewConstraintClass<T> where T : new()
    {
    }

    public class TypeParameterConstraintClass<T> where T : SomeInterface
    {
    }

    private class MultipleTypeParameterConstraintsClass<T1, T2> where T1 : SomeInterface, SomeInterface2, new()
        where T2 : SomeInterface2
    {
    }

    class IndexClass
    {
        private bool a;

        public bool this[int index]
        {
            get { return a; }
            set { a = value; }
        }
    }

}

public interface SomeInterface
{
}

internal interface SomeInterface2
{
}